cache/test_cache.py

```diff
@@ -1734,4 +1735,124 @@ def test_http1(self):
         self.test_h2()
 
 
+@parameterize_class(
+    [
+        {"name": "Http", "clients": [DEPROXY_CLIENT]},
+        {"name": "H2", "clients": [DEPROXY_CLIENT_H2]},
+    ]
+)
+class TestCacheResponseWithCache(tester.TempestaTest):
+    """This class contains checks for tempesta cache config."""
+
+    tempesta = {
+        "config": """
+listen 80;
+listen 443 proto=h2;
+
+server ${server_ip}:8000;
+cache 2;
+cache_methods GET HEAD;
+cache_fulfill * *;
+
+tls_match_any_server_name;
+
+vhost default {
+    proxy_pass default;
+    tls_certificate ${tempesta_workdir}/tempesta.crt;
+    tls_certificate_key ${tempesta_workdir}/tempesta.key;
+}
+""",
+    }
+
+    backends = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": "HTTP/1.1 200 OK\r\nConnection: keep-alive\r\nContent-Length: 0\r\n\r\n",
+        }
+    ]
+
+    def encode_chunked(self, data, chunk_size=256):
+        result = ""
+        while len(data):
+            chunk, data = data[:chunk_size], data[chunk_size:]
+            result += f"{hex(len(chunk))[2:]}\r\n"
+            result += f"{chunk}\r\n"
+        return result + "0\r\n\r\n"
+
+    @parameterize.expand(
+        [
+            param(name="GET", method="GET", trailers_expected=True),
+            param(name="HEAD", method="HEAD", trailers_expected=False),
+        ]
+    )
+    def test(self, name, method, trailers_expected):
+        self.start_all_services()
+
+        srv: StaticDeproxyServer = self.get_server("deproxy")
+        srv.set_response(
+            "HTTP/1.1 200 OK\r\n"
+            + "Content-type: text/html\r\n"
+            + f"Last-Modified: {deproxy.HttpMessage.date_time_string()}\r\n"
+            + f"Date: {deproxy.HttpMessage.date_time_string()}\r\n"
+            + "Server: Deproxy Server\r\n"
+            + "Transfer-Encoding: chunked\r\n"
+            + "Trailer: X-Token1 X-Token2\r\n\r\n"
+            + self.encode_chunked(string.ascii_letters, 16)[:-2]
+            + f"X-Token1: value1\r\n"
+            + f"X-Token2: value2\r\n\r\n"
+        )
+
+        client = self.get_client("deproxy")
+        request = client.create_request(method="GET", headers=[])
+        client.send_request(request, "200")
+
+        if isinstance(self, TestCacheResponseWithCacheH2):
+            self.assertFalse(client.last_response.headers.get("Trailer"))
+            self.assertFalse(client.last_response.headers.get("Transfer-Encoding"), "chunked")
+        else:
+            self.assertTrue(client.last_response.headers.get("Trailer"), "X-Token1 X-Token2")
+            self.assertTrue(client.last_response.headers.get("Transfer-Encoding"), "chunked")
+        self.assertFalse(client.last_response.headers.get("X-Token1"))
+        self.assertFalse(client.last_response.headers.get("X-Token2"))
```

Comment:
```suggestion
            self.assertIsNone(client.last_response.headers.get("Trailer"))
            self.assertNotEqual(client.last_response.headers.get("Transfer-Encoding"), "chunked")
        else:
            self.assertEqual(client.last_response.headers.get("Trailer"), "X-Token1 X-Token2")
            self.assertEqual(client.last_response.headers.get("Transfer-Encoding"), "chunked")
        self.assertIsNone(client.last_response.headers.get("X-Token1"))
        self.assertIsNone(client.last_response.headers.get("X-Token2"))
```

Please use the correct asserts. This will make the output more informative.